scripts/west_commands/runners/nrfjprog.py

```diff
@@ -151,10 +167,24 @@ def do_run(self, command, **kwargs):
                 program_cmd
             ])
         else:
-            if self.family == 'NRF52':
+            if self.family == 'NRF51':
+                commands.append(program_cmd + ['--sectorerase'])
+            elif self.family == 'NRF52':
                 commands.append(program_cmd + ['--sectoranduicrerase'])
             else:
-                commands.append(program_cmd + ['--sectorerase'])
+                uicr = {
+                    'NRF53': ((0x00FF8000, 0x00FF8800),
+                              (0x01FF8000, 0x01FF8800)),
+                    'NRF91': ((0x00FF8000, 0x00FF8800),),
+                }[self.family]
+
+                if not self.force and has_region(uicr, self.hex_):
+                    # Hex file has UICR contents.
+                    raise RuntimeError('The hex file contains data placed in '
+                                       'the UICR, which needs a full erase. '
+                                       'Run west flash again with --erase.')
```

Comment:
Thanks for adding that, can you please update the error here then too?

```suggestion
                                       'Run west flash again with --force or --erase.')
```

Reply:
Done